### PR TITLE
feat(categories): add filter for tag list

### PR DESCRIPTION
**Pull request by donni106**
SVAK-51

### CODE REVIEW

File: app/graphql/resolvers/categories_search.rb [hunk 1]
```diff
@@ -43,6 +44,10 @@ def apply_exclude_ids(scope, value)
     scope.where.not(id: value)
   end
 
+  def apply_tag_list(scope, value)
+    scope.tagged_with(value, any: true)
```
Comment (marcometz) [inline app/graphql/resolvers/categories_search.rb:48]: if it should be any tag or all tags matching could be different in each usecase.
Could we make it configurable if it is any: or all: ?

**Review by marcometz — APPROVED** [review]

File: app/graphql/resolvers/categories_search.rb (hunk 1 above)
Comment (donni106) [reply app/graphql/resolvers/categories_search.rb:48]: oh good point, i will check if this is something quick.

Comment (donni106) [reply app/graphql/resolvers/categories_search.rb:48]: exclude is also a valid option. i will integrate all three possibilities:

```
# Find users that matches all given tags:
# NOTE: This only matches users that have the exact set of specified tags. If a user has additional tags, they are not returned.
User.tagged_with(["awesome", "cool"], :match_all => true)

# Find users with any of the specified tags:
User.tagged_with(["awesome", "cool"], :any => true)

# Find users that have not been tagged with awesome or cool:
User.tagged_with(["awesome", "cool"], :exclude => true)
```
https://github.com/mbleigh/acts-as-taggable-on?tab=readme-ov-file#finding-tagged-objects

File: app/graphql/resolvers/categories_search.rb [hunk 2]
```diff
@@ -43,6 +57,15 @@ def apply_exclude_ids(scope, value)
     scope.where.not(id: value)
   end
 
+  def set_tagging_strategy(scope, value)
+    @tagging_strategy = value
+    scope
+  end
+
+  def apply_tag_list(scope, value)
+    scope.tagged_with(value, TAGGING_STRATEGIES[@tagging_strategy])
```
Comment (marcometz) [inline app/graphql/resolvers/categories_search.rb:66]: hier fehlt eventuell noch ein Fallback falls taggingStrategy nicht definiert ist?

**Pull request merged**